src/components/TaskList/index.jsx

```diff
@@ -43,7 +64,7 @@ const TaskList = ({ tasks, limit, filter, setTaskLength, isCardCreated }) => {
     isCardCreated
       ? dispatch(paginationLimitUpdate(PAGINATION_LIMIT - 1))
       : dispatch(paginationLimitUpdate(PAGINATION_LIMIT));
-  }, [filter]);
+  }, [filterState]);
 
   return filteredTasks
     .slice(0, limit)
```

Comment:
```suggestion
const paginatedTasks = filteredTasks .slice(0, limit);

  return paginatedTasks
```
[reference](https://github.com/azo-wtag/todo-app-react/pull/6#discussion_r1186783525)